Java code generation broken with maven > 3.0.x

This is a problem with one of the plugins not supporting newer maven APIs. The workaround is to build the jar with maven 3.0.x.

[ERROR] Failed to execute goal com.webguys:string-template-maven-plugin:1.1:render (default) on project riak-pb: Execution default of goal com.webguys:string-template-maven-plugin:1.1:render failed: An API incompatibility was encountered while executing com.webguys:string-template-maven-plugin:1.1:render: java.lang.NoSuchMethodError: org.apache.maven.execution.MavenSession.getRepositorySession()Lorg/sonatype/aether/RepositorySystemSession;

--- a/pom.xml
+++ b/pom.xml
@@ -78,6 +78,13 @@
                         </goals>
                     </execution>
                 </executions>
+                <dependencies>
+                    <dependency>
+                        <groupId>org.twdata.maven</groupId>
+                        <artifactId>mojo-executor</artifactId>
+                        <version>2.2.0</version>
+                    </dependency>
+                </dependencies>
             </plugin>
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>

